R/SS_readctl_3.24.R

```diff
@@ -49,40 +43,23 @@ SS_readctl_3.24 <- function(file,
                             # This information is needed if Q_type of 3 or 4 is used
                             N_CPUE_obs = NULL,
                             ##################################
-                            ptype = FALSE) {
+                            ptype = lifecycle::deprecated()) {
   # deprecated variable warnings -----
   # soft deprecated for now, but fully deprecate in the future.
-  if (lifecycle::is_present(echoall)) {
+  if (lifecycle::is_present(ptype)) {
     lifecycle::deprecate_warn(
-      when = "1.41.1",
-      what = "SS_readctl_3.24(echoall)"
-    )
-  }
-  if (lifecycle::is_present(version)) {
-    lifecycle::deprecate_warn(
-      when = "1.41.1",
-      what = "SS_readctl_3.24(version)"
+      when = "1.45.0",
+      what = "SS_readctl_3.24(ptype)"
     )
   }
 
-  version <- "3.24"
-  if (lifecycle::is_present(Ngenders)) {
-    lifecycle::deprecate_warn(
-      when = "1.41.1",
-      what = "SS_readctl_3.24(Ngenders)",
-      details = "Please use Nsexes instead. Ability to use Ngenders will be dropped in next release."
-    )
-    Nsexes <- Ngenders
-  }
-
-
   # internally used fun definitions ----
   # function to read Stock Synthesis data files
 
   if (verbose) message("running SS_readctl_3.24\n")
   dat <- readLines(file, warn = FALSE)
 
-  nver <- as.numeric(substring(version, 1, 4))
+  nver <- as.numeric(substring("3.24", 1, 4))
```

Comment:
> Thanks @k-doering-NOAA for these changes. I have a few changes that I think should be implemented (see individual comments), I pushed one formatting commit of the warning for PYype found in a parameter data frame, I found 24 instances of Ptype in [SSplotPars()](https://github.com/r4ss/r4ss/blob/bd7ca77dab0827e7e887d766e27a87768231b516/R/SSplotPars.R#L102) that I think should be removed, I think that [this line in SS_readdat_3.24](https://github.com/r4ss/r4ss/blob/44cf139540e8d7e13fe89c69a05f8f7a55d2959e/R/SS_readdat_3.24.R#L503) might need to use `Nsexes` instead of `Ngenders`, do we want to get rid of `Ngenders` in [`SS_readdat_3.30()`](https://github.com/r4ss/r4ss/blob/7bb486f86958c6da5d7a228c0565792cdc63b6cb/R/SS_readdat_3.30.R#L197-L199).

Thanks, Kelli for all the comments! Good catches on a lot of these, I will make the changes shortly.

The only one I think that doesn't need to be changed are the 24 instances of Ptype in SSplotPars - confusingly, this Ptype is actually representing something different (prior types, rather than the Parameter type as in the read/write functions). I think it was just an accidental collision of names that happened when 2 people wrote different functions